lib/middlewares/route.js

```diff
@@ -30,7 +30,10 @@ module.exports = function(app) {
       return;
     }
 
-    res.setHeader('Content-Type', extname ? mime.lookup(extname) : 'application/octet-stream');
+    // Do not overwrite header if already set by another middleware
+    if (!res.getHeader('Content-Type')) {
```

Comment:
Please use `hasHeader` to detect whether a header has been set and add test for the new behaviors. Otherwise it looks good to me.

Reply:
Done, though `hasHeader` [will not work](https://nodejs.org/api/http.html#http_response_hasheader_name) with Node < 7.7.0